### ISSUE TITLE
No reconnection after PRIMARY down if there is a model.watch

### ISSUE THREAD

**Issue by majordo** [Automattic/mongoose#6741]
I am using mongoose@5.2.4 with a replicaSet of 3 nodes with mongo v3.6.2.
Node v8.11.2
**Do you want to request a *feature* or report a *bug*?** BUG

I noticed that reconnection does not work if there is a model.watch.
If I enable model.watch and turn off the PRIMARY I get this error:
```javascript
.../node_modules/mongodb/lib/utils.js:132
      throw err;
      ^
MongoNetworkError: connection 6 to 192.168.0.1:27017 closed
```
If there are no model.watch there are no errors and my application normally reconnects after the election of the PRIMARY node.

**If the current behavior is a bug, please provide the steps to reproduce.**
```javascript
import mongoose from "mongoose";
const uri="mongodb://192.168.0.1:27017,192.168.0.2:27017,192.168.0.3:27017/test?replicaSet=test"
mongoose.connect(config.mongodb.uri, {
    useNewUrlParser: true,
});
const mongoDB = mongoose.connection;
mongoDB
    .on('connecting', () => console.log('connecting to MongoDB...'))
    .on('connected', () => console.log('MongoDB connected...'))
    .on('reconnected', () => console.log('MongoDB reconnected!'))
    .on('disconnecting', () => console.log('MongoDB disconnecting...'))
    .on('timeout', (e) => console.log('MongoDB timeout ->', e))
    .on('disconnected', () => console.log('MongoDB disconnected'))
    .on('error', () => {
         mongoose.disconnect();
         console.log('Failed to connect to MongoDB.')
    })
    .once('open', () => console.log('MongoDB connection opened! DONE!'))


// userModel
//     .watch(null, { fullDocument: "updateLookup" })
//     .on("change", change => {
//         console.warn('USER CHANGE:', change)
//     })

```

**Comment by vkarpov15** [comment]
Are you handling the error event from change streams? `Model.watch().on('change', changeHandler).on('error', handleError)`

**Comment by majordo** [comment]
No. Just the code in my previous post it's enough for generating the error...

**Comment by majordo** [comment]
Hello,
it's my clarification enough? Have you been able to reproduce the error? Can I help somehow?

**Comment by vkarpov15** [comment]
Nope, we'll need to investigate and see if we can reproduce this.

**Comment by vkarpov15** [comment]
Unfortunately this is a tricky issue to handle, we'll have to push this back to a minor release. Here's a workaround for now, we'll need to add docs that change streams emit an 'error' event.

```javascript
const assert = require('assert');
const mongoose = require('mongoose');
mongoose.set('debug', true);

const GITHUB_ISSUE = `gh6741`;
const connectionString = `mongodb://localhost:27017/${ GITHUB_ISSUE }`;
const { Schema } = mongoose;

run().then(() => console.log('done')).catch(error => console.error(error.stack));

async function run() {
  await mongoose.connect(connectionString, { useNewUrlParser: true, replicaSet: 'rs' });
  await mongoose.connection.dropDatabase();

  await mongoose.connection.createCollection('Foo');

  const M = mongoose.model('Foo', new Schema({ name: String }), 'Foo');

  let changeStream = M.watch().on('change', change => console.log(change));
  changeStream.on('error', handleChangeStreamError);
  console.log('Watching...');

  function handleChangeStreamError(error) {
    console.log('Got Error', error.toString());
    if (error.name !== 'MongoNetworkError') {
      throw error;
    }
    changeStream = M.watch().on('change', change => console.log(change)).on('error', handleChangeStreamError);
  }
}
```

Output from above script:

```
$ node gh-6741.js 
Mongoose: Foo.watch()
Watching...
done
Got Error MongoNetworkError: connection 1 to localhost:27018 closed
Mongoose: Foo.watch()
Got Error MongoNetworkError: connection 0 to localhost:27017 closed
Mongoose: Foo.watch()
```

**Comment by vkarpov15** [comment]
Follow https://jira.mongodb.org/browse/NODE-1617 for updates on progress

**Comment by vkarpov15** [comment]
Fixed by #6920
